src/runtime/Loader.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { Component } from "react";
+
+interface Props {
+  render: JSX.Element | undefined;
+}
+export const LoaderContext = React.createContext(undefined);
+
+export class Loader extends Component<Props> {
+  public state = {
+    render: undefined,
+  };
+
+  public componentDidMount() {
+    console.log(this.props.render);
+    if (this.props.render) {
+      this.setState({ render: this.props.render });
```

Comment:
Do we need to have this in the state since it's already in the props? I think this is redundant and unneeded, and will just result in an extra unnecessary rerender.